Content.Server/StationEvents/Metric/CombatMetric.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Content.Server.Chemistry.EntitySystems;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Mind.Components;
+using Content.Server.StationEvents.Metric.Components;
+using Content.Shared.Damage;
+using Content.Shared.FixedPoint;
+using Content.Shared.Fluids.Components;
+using Content.Shared.Humanoid;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Zombies;
+
+namespace Content.Server.StationEvents.Metric;
+
+/// <summary>
+///   Measures the strength of friendies and hostiles. Also calculates related health / death stats.
+///
+///   I've used 10 points per entity because later we might somehow estimate combat strength
+///   as a multiplier. We could for instance detect damage delt / recieved and look also at
+///   entity hitpoints & resistances as an analogue for danger.
+///
+///   Writes the following
+///   Friend : -10 per each friendly entity on the station (negative is GOOD in chaos)
+///   Hostile : about 10 points per hostile (those with antag roles) - varies per constants
+///   Combat: friendlies + hostiles (to represent the balance of power)
+///   Death: 20 per dead body,
+///   Medical: 10 for crit + 0.05 * damage (so 5 for 100 damage),
+/// </summary>
+public sealed class CombatMetric : ChaosMetricSystem<CombatMetricComponent>
+{
+    public override ChaosMetrics CalculateChaos(EntityUid metric_uid, CombatMetricComponent component, ChaosMetricComponent metric,
+        CalculateChaosEvent args)
+    {
+        // Add up the pain of all the puddles
+        var query = EntityQueryEnumerator<MindComponent, MobStateComponent, DamageableComponent>();
+        FixedPoint2 hostiles = 0.0f;
+        FixedPoint2 friendlies = 0.0f;
+
+        FixedPoint2 medical = 0.0f;
+        FixedPoint2 death = 0.0f;
+
+        var nukieQ = GetEntityQuery<NukeOperativeComponent>();
+        var zombieQ = GetEntityQuery<ZombieComponent>();
+
+        var humanoidQ = GetEntityQuery<HumanoidAppearanceComponent>();
```

Comment:
This looks copied again.

Also gameevents shouldn't be hardcoded in here and needs splitting out individually.